utils/docker-compose/scripts/teardown.sh

```diff
@@ -8,19 +8,41 @@
 set -e
 
 export SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
-export COMPOSE_PROJECT_NAME=fabric-fpc
-export NETWORK_CONFIG=${SCRIPT_DIR}/../network-config
 
-# Shut down the Docker containers for the system tests.
-docker-compose -f $NETWORK_CONFIG/docker-compose.yml stop
-docker-compose -f $NETWORK_CONFIG/docker-compose.yml kill && docker-compose -f $NETWORK_CONFIG/docker-compose.yml down
+. ${SCRIPT_DIR}/lib/common.sh
 
-# remove the local state
-rm -f ~/.hfc-key-store/*
 
-# remove chaincode docker images
-docker rm $(docker ps -aq)
-docker rmi $(docker images dev-* -q)
+# Shut down the Docker containers for the system tests and remove temporary volumes
+${DOCKER_COMPOSE} stop
+${DOCKER_COMPOSE} kill && ${DOCKER_COMPOSE} down
 
-# Your system is now clean
-docker volume prune
+if [ "$1" == '--clean-slate' ]; then
+        echo "removing state of CA etc."
+	${DOCKER_COMPOSE} kill && ${DOCKER_COMPOSE} down --volumes
+
+	# CA state got destroyed, so corresponding wallets are obsolete
+        if [ -d "${NODE_WALLETS}" ]; then
+		echo "deleting obsolet wallets in '${NODE_WALLETS}'"
+		rm -rf "${NODE_WALLETS}";
+	fi
```

Comment:
Nope, a down without --volumes option does not make wallets obsolete as the CA retains it's keys on an anonymous volume. Only down --volumes removes this . (e.g., otherwise you also would have to re-run generate after each teardown)